dashboard/config/routes.rb

```diff
@@ -754,4 +754,9 @@
   get '/form/:misc_form_path', to: 'foorm/misc_survey#new'
 
   post '/i18n/track_string_usage', action: :track_string_usage, controller: :i18n
+
+  namespace :foorm do
+    post 'form/(*name)/:version', action: :create, controller: :forms
+    put 'form/(*name)/:version', action: :update, controller: :forms
+  end
```

Comment:
this whole block is a bit unconventional so I have a few questions :-) They are somewhat inter-related so it might be worth discussing before making a bunch of code updates:
1. how is the client planning to use the POST API, more specifically how will they know what version to specify? if the client needs the next unused version number for a given form name, then a more conventional approach would be `post 'form/(*name)'`, and the server would provide the next unused version id. Alternately, if the client knows the version number and wants to overwrite it regardless of whether it exists, then you can use the `update` route and you wouldn't need the `create` route at all.
2. would `form/:name` work here instead of `*name`? The uses of the `*name` syntax seem a bit complicated: https://guides.rubyonrails.org/routing.html so I would suggest only using them if we have a specific reason we need them.
3. if possible it would be better to use rails `resources`, but it's hard to say how just yet depending on some of the other answers.

Reply:
Discussed offline. Made this [PR](https://github.com/code-dot-org/code-dot-org/pull/37887) to keep the ids stable so we can have better routes here and in the future.

Reply:
updated with new routes (see comment on overall PR)